pkg/controllers/provisioning/batcher.go

```diff
@@ -57,26 +59,23 @@ func (b *Batcher) TriggerImmediate() {
 	b.immediate <- struct{}{}
 }
 
-// Wait starts a batching window and returns a slice of items when closed.
-func (b *Batcher) Wait() (window time.Duration) {
+// WaitAndInjectSettings starts a batching window and injects settings into the existing context
+func (b *Batcher) WaitAndInjectSettings(ctx context.Context) context.Context {
```

Comment:
Seems like this shouldn't be the batcher's job. At one point, we were considering abstracting this for other batching purposes. Does it make sense for the top level provisioning loop own the top level context?